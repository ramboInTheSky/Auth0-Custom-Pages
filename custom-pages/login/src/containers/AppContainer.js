import React from 'react'
import PropTypes from 'prop-types'
import { Router } from 'react-router'

class AppContainer extends React.Component {
  static defaultProps = {
    routerKey: null,
  }
  static propTypes = {
    history: PropTypes.shape().isRequired,
    routes: PropTypes.shape().isRequired,
    routerKey: PropTypes.number,
  }

  render() {
    const { history, routes, routerKey } = this.props

    return (
      <div>
        <Router history={history} children={routes} key={routerKey} />
      </div>
    )
  }
}
export default AppContainer
