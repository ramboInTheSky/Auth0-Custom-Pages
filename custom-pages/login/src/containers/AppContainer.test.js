import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-14'

import AppContainer from './AppContainer'

Enzyme.configure({ adapter: new Adapter() })

const props = { history: {}, routes: {}, routerKey: 1 }

describe('AppContainer components', () => {
  it('should render <AppContainer /> correctly', () => {
    expect(shallow(<AppContainer {...props} />)).toMatchSnapshot()
  })
})
