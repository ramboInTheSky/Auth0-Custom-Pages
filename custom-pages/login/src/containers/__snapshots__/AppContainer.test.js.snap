// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer components should render <AppContainer /> correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppContainer
    history={Object {}}
    routerKey={1}
    routes={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Router
        history={Object {}}
        render={[Function]}
      >
        Object {}
      </Router>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": "1",
      "nodeType": "class",
      "props": Object {
        "children": Object {},
        "history": Object {},
        "render": [Function],
      },
      "ref": null,
      "rendered": Object {},
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Router
          history={Object {}}
          render={[Function]}
        >
          Object {}
        </Router>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": Object {},
          "history": Object {},
          "render": [Function],
        },
        "ref": null,
        "rendered": Object {},
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFourteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
